src/distraction_manager.cpp

```diff
@@ -147,72 +132,16 @@ void distraction_manager_gui::show()
-        if( currentAction == "QUIT" ) {
+        if( action == "QUIT" ) {
             break;
         }
 
-        if( currentAction == "UP" ) {
-            if( currentLine > 0 ) {
-                --currentLine;
-            } else {
-                currentLine = number_of_distractions - 1;
-            }
-        } else if( currentAction == "DOWN" ) {
-            if( currentLine == number_of_distractions - 1 ) {
-                currentLine = 0;
-            } else {
-                ++currentLine;
-            }
-        } else if( currentAction == "CONFIRM" ) {
-            // This will change status color and status text
-            distractions_status[currentLine] = !distractions_status[currentLine];
-
-            // This will actually toggle enabled/disabled status of distractions
-            switch( currentLine ) {
-                case 0:
-                    uistate.distraction_noise = !uistate.distraction_noise;
-                    break;
-                case 1:
-                    uistate.distraction_pain = !uistate.distraction_pain;
-                    break;
-                case 2:
-                    uistate.distraction_attack = !uistate.distraction_attack;
-                    break;
-                case 3:
-                    uistate.distraction_hostile_close = !uistate.distraction_hostile_close;
-                    break;
-                case 4:
-                    uistate.distraction_hostile_spotted = !uistate.distraction_hostile_spotted;
-                    break;
-                case 5:
-                    uistate.distraction_conversation = !uistate.distraction_conversation;
-                    break;
-                case 6:
-                    uistate.distraction_asthma = !uistate.distraction_asthma;
-                    break;
-                case 7:
-                    uistate.distraction_dangerous_field = !uistate.distraction_dangerous_field;
-                    break;
-                case 8:
-                    uistate.distraction_weather_change = !uistate.distraction_weather_change;
-                    break;
-                case 9:
-                    uistate.distraction_hunger = !uistate.distraction_hunger;
-                    break;
-                case 10:
-                    uistate.distraction_thirst = !uistate.distraction_thirst;
-                    break;
-                case 11:
-                    uistate.distraction_temperature = !uistate.distraction_temperature;
-                    break;
-                case 12:
-                    uistate.distraction_mutation = !uistate.distraction_mutation;
-                    break;
-                default:
-                    return;
-            }
+        if( navigate_ui_list( action, currentLine, 5, number_of_distractions, true ) ) {
+        } else if( action == "CONFIRM" ) {
+            *std::get<0>( configurable_distractions[currentLine] ) = !*std::get<0>
+                    ( configurable_distractions[currentLine] );
```

Comment:
Should I put it in a local reference? Or just `^=1` it? Not sure of our style preferences for something like this.